### ISSUE TITLE
How does setOptions() work?

### ISSUE THREAD

**Issue by patrickmcelwee** [grtjn/ml-visjs-graph.js#27]
When I create a graph like this:

```javascript
  createGraph() {
    this.mlVisjsGraph = new mlvisjs.Graph(
      this.container.current,
      null, // templateUri
      {
        get: () => template // templateCache
      },
      graph => {
        if (this.props.data) {
          graph.network.setData(
            this.props.data.nodes,
            null,
            this.props.data.edges,
            null
          );
        }

        if (this.props.events) {
          graph.network.setEvents(this.props.events);
        }

        if (this.props.options) {
          graph.network.setOptions(this.props.options);
        }
      }
    );
  }
```

I can confirm that I am passing in `this.props.options`, but it does not affect my graph. I was just wondering if you have seen this work before. It is very possible that I am doing something wrong.

**Comment by grtjn** [comment]
I have this in Vue:

```javascript
  mounted() {
    var self = this;

    if (!mlvisjs) {
      throw 'Error: mlvisjs not found, required by mlVisjsGraph directive';
    }

    new mlvisjs.Graph(self.$el, null, null, function done(graph) {
      if (self.nodes.length || self.edges.length) {
        graph.network.setData(self.cleanNodes, null, self.cleanEdges, null);
      }
      if (self.events) {
        graph.network.setEvents(self.cleanEvents);
      }
      if (self.options) {
        graph.network.setOptions(self.cleanOptions);
      }
      if (self.layout) {
        graph.setLayout(self.layout);
      }
      if (self.styling) {
        graph.setStyling(self.styling);
      }
      self.graph = graph;
      self.sleep(100).then(() => {
        if (self.options) {
          graph.network.setOptions(self.cleanOptions);
        }
      });
    }, function fail(failure) {
      throw failure;
    });
  }
```

It looks pretty similar, but I do keep a ref to `this` in a var `self`, because sometimes `this` does not point back to the component, even though you might expect it to..

**Comment by patrickmcelwee** [comment]
Thanks! 

So, I'm not sure what I changed (actually, git tells me I didn't really change much) ... but after fiddling for a bit, now it is working for me. (head-scratcher)

One thing is that I couldn't use the simple `{edges: { color: 'red' } }`, because the color property gets overwritten. I had to nest it further: `{edges: { color: { color: 'red } } }`. But that doesn't explain why my nodes options started working ... (shrug)

I notice you have a sleep before setting options. I found I don't need that, FYI.
